src/api/schema/broker.rs

```diff
@@ -0,0 +1,67 @@
+use futures::channel::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use futures::task::{Context, Poll};
+use futures::{Stream, StreamExt};
+use once_cell::sync::Lazy;
+use serde::export::PhantomData;
+use slab::Slab;
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+use std::pin::Pin;
+use std::sync::Mutex;
+
+/// Current subscriptions. This is static so that 'global' changes can be published to any
+/// connecting subscriber. Subscribers fall out of scope once the subscription is terminated
+static SUBSCRIBERS: Lazy<Mutex<HashMap<TypeId, Box<dyn Any + Send>>>> = Lazy::new(Default::default);
+
+struct Senders<T>(Slab<UnboundedSender<T>>);
```

Comment:
Do you think we could use bounded channels here?

Reply:
👍 Switched to bounded chans, with a size of 20. I think this should be enough to keep back-pressure low but still feel real-time for a reasonable # of connected clients.

Updated [the client to match](https://github.com/timberio/vector/pull/4581/commits/c9316ce38590ec4c01854ab5e958f00df271696b#diff-88d0e464468fdbce20a25a363596ec78830b07a1bbe556270e60ff948ffb71f3R162); previously it was 100, which was a bit high.